### PR TITLE
Various fixes to item frame rollbacks

### PR DESCRIPTION
**Pull request by PhanaticD**

### CODE REVIEW

File: src/main/java/me/botsko/prism/actions/HangingItemAction.java [hunk 1]
```diff
@@ -105,28 +105,28 @@ public ChangeResult applyRestore(Player player, QueryParameters parameters, bool
...
-        if( me.botsko.elixr.BlockUtils.materialMeansBlockDetachment( loc.getBlock().getType() ) ) { return new ChangeResult(
+        //Ensure there's a block at this location that accepts an attachment
+        if( me.botsko.elixr.BlockUtils.materialMeansBlockDetachment( loc.getBlock().getRelative( getDirection() ).getLocation().getBlock().getType() ) ) { return new ChangeResult(
                 ChangeResultType.SKIPPED, null ); }
 
         try {
             if( getHangingType().equals( "item_frame" ) ) {
                 final Hanging hangingItem = getWorld().spawn( loc, ItemFrame.class );
-                hangingItem.setFacingDirection( attachedFace, true );
+                hangingItem.setFacingDirection( attachedFace, true );
```
Comment (viveleroi) [inline src/main/java/me/botsko/prism/actions/HangingItemAction.java:119]: Can you trim these empty spaces at the end of the lines please?

File: src/main/java/me/botsko/prism/database/mysql/SelectQueryBuilder.java [hunk 2]
```diff
@@ -330,7 +330,7 @@ protected String group() {
     @Override
     protected String order() {
         final String sort_dir = parameters.getSortDirection();
-        return " ORDER BY " + tableNameData + ".epoch " + sort_dir + ", x ASC, z ASC, y ASC, id " + sort_dir;
+        return " ORDER BY " + tableNameData + ".epoch " + sort_dir + ", id " + sort_dir;
```
Comment (viveleroi) [inline src/main/java/me/botsko/prism/database/mysql/SelectQueryBuilder.java:333]: Have you fully tested these changes? We're sorting by coordinate, most importantly y, so we don't put a falling block above blocks which haven't yet been put back. Same with attached blocks like torches and such. There's no guarantee that records will be recorded in the right order.

**Review by viveleroi — CHANGES_REQUESTED** [review]

File: src/main/java/me/botsko/prism/database/mysql/SelectQueryBuilder.java (hunk 2 above)
Comment (PhanaticD) [reply src/main/java/me/botsko/prism/database/mysql/SelectQueryBuilder.java:333]: i will look into this more and find something that works with both

**Comment by Narimm** [comment]
Given the age of this PR we might just close it off @PhanaticD if you want to resubmit feel free :P

**Pull request closed without merging**